src/utils/layout-helpers.ts

```diff
@@ -6,9 +6,14 @@ import { clean, coerce, valid } from 'semver';
 import { BaseProject } from '../base-project';
 import { fsProvider } from '../fs-provider';
 import walkAsync from './walk-async';
+import { instrumentTime } from './logger';
 
 // const GLOBAL_REGISTRY = ['primitive-name'][['relatedFiles']];
 
+// Don't traverse dependencies we've already seen.
+// correct package graph can sort of throw us in to cycles if we don't keep track of this.
+const SEEN = new Set<string>();
```

Comment:
seems we need to specify `SEEN` as additional argument for `getRecursiveInRepoAddonRoots`, because we may have multiple ember apps & addons per language server. 
having it at top-level will break 1+ app/addon discovery

Reply:
perhaps -- in monorepos we need the information shared though.

Reply:
if SEEN will be filled-up by one project, another will be unable to count it as it's own too.